java/modules/core/src/main/java/org/apache/synapse/mediators/transform/url/RewriteRule.java

```diff
@@ -53,7 +55,7 @@ public void rewrite(URIFragments fragments,
                         MessageContext messageContext) throws URISyntaxException {
 
         if (condition != null) {
-            String uriString = fragments.toURIString();
+            @StartsWith({"https", "file"}) String uriString = fragments.toURIString();
```

Comment:
Why do you need this local variable annotation? It should be inferred if there is no need to suppress a warning.